pkg/interceptors/github/github.go

```diff
@@ -96,7 +126,270 @@ func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequ
 		}
 	}
 
+	// For event types pull_request, issue_comment check github owners approval
+	if p.OwnersCheckEnabled {
+		ownerCheckAllowed := false
+		for _, allowedEvent := range ownersEventTypes {
+			if actualEvent == allowedEvent {
+				ownerCheckAllowed = true
+				break
+			}
+		}
+		if ownerCheckAllowed {
+			ns, _ := triggersv1.ParseTriggerID(r.Context.TriggerID)
+			ghToken, err := w.SecretGetter.Get(ctx, ns, p.GithubToken)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error getting github token: %v", err)
+			}
+			client := makeClient(ctx, headers.Get("X-Github-Enterprise-Host"), string(ghToken))
```

Comment:
Will this work for the public GitHub? Or is only Enterprise supported?

Reply:
It supports both public and enterprise. If enterprise host is not provided then the client would generate the `NewClient` instead of `NewEnterpriseClient`. After rasing PR I have started to change this makeClient to be much more robost. I will be updating it